src/services/pokemon.js

```diff
@@ -0,0 +1,34 @@
+export async function fetchPokedex() {
+  const resp = await fetch('https://pokedex-alchemy.herokuapp.com/api/pokedex');
+  const data = await resp.json();
+  return data.results;
+}
+
+export async function fetchTypes() {
+  const resp = await fetch('https://pokedex-alchemy.herokuapp.com/api/pokedex/types');
+  const data = await resp.json();
+  return data.map((item) => item.type);
+}
+
+export async function fetchFilteredPokemon(selectedType, search) {
+  const params = new URLSearchParams();
+  if (selectedType !== 'All') {
+    params.set('type', selectedType);}
+  if (search) {
+    params.set('pokemon', search);
+  }
+  const resp = await fetch(`https://pokedex-alchemy.herokuapp.com/api/pokedex?${params.toString()}`);
+  const data = await resp.json();
+  return data.results;
+}
+
+export async function fetchSortedPokemon(option) {
+  const params = new URLSearchParams();
+  if (option === 'A-Z') {
+    params.set('asc', option);
+  } 
+  if (option === 'Z-A') { params.set('desc', option);}
+  const resp = await fetch(`https://pokedex-alchemy.herokuapp.com/api/pokedex?sort=pokemon&direction=${params.toString()}`);
+  const data = await resp.json();
+  return data.results;
+}
```

Comment:
In order for our search function to work on filtered pokemon and search pokemon, we would want to add in the option params to the fetchFilteredPokemon function. And remember, we want the params.set value to be ('direction', 'asc' or 'desc'), which would look like direction=asc in our url when params.toString() is called.